src/main/java/androidx/widget/TextView.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.widget
+
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.TextView
+
+/**
+ * Add an action which will be invoked after text of this TextView has been changed.
+ *
+ * @return the [TextWatcher] added to the TextView
+ * @see [TextWatcher.afterTextChanged]
+ */
+fun TextView.doAfterChanged(
+    afterChanged: (Editable) -> Unit
+): TextWatcher = addTextChangedListener(afterChanged = afterChanged)
+
+/**
+ * Add an action which will be invoked before text of this TextView has been changed.
+ *
+ * @return the [TextWatcher] added to the TextView
+ * @see [TextWatcher.beforeTextChanged]
+ */
+fun TextView.doBeforeChanged(
+    beforeChanged: (s: CharSequence, start: Int, count: Int, after: Int) -> Unit
+): TextWatcher = addTextChangedListener(beforeChanged = beforeChanged)
+
+/**
+ * Add an action which will be invoked when text of this TextView has been changed.
+ *
+ * @return the [TextWatcher] added to the TextView
+ * @see [TextWatcher.onTextChanged]
+ */
+fun TextView.doOnChanged(
```

Comment:
I feel these methods should be called `doOnTextChanged`, `doBeforeTextChanged` and `doAfterTextChanged` for clarity and consistency with `addTextChangedListener` and the methods on `TextWatcher`.

Reply:
Makes sense. I am going to update PR soon.